translib/transformer/test/utils_test.go

```diff
@@ -0,0 +1,226 @@
+//  distributed under the License is distributed on an "AS IS" BASIS,         //
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  //
+//  See the License for the specific language governing permissions and       //
+//  limitations under the License.                                            //
+//                                                                            //
+////////////////////////////////////////////////////////////////////////////////
+
+package transformer_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"reflect"
+	"testing"
+
+	. "github.com/Azure/sonic-mgmt-common/translib"
+	db "github.com/Azure/sonic-mgmt-common/translib/db"
+	"github.com/go-redis/redis/v7"
+)
+
+func checkErr(t *testing.T, err error, expErr error) {
+	if err.Error() != expErr.Error() {
+		t.Errorf("Error %v, Expect Err: %v", err, expErr)
+	} else if reflect.TypeOf(err) != reflect.TypeOf(expErr) {
+		t.Errorf("Error type %T, Expect Err Type: %T", err, expErr)
+	}
+}
+
+func processGetRequest(url string, expectedRespJson string, errorCase bool, expErr ...error) func(*testing.T) {
+	return func(t *testing.T) {
+		var expectedMap map[string]interface{}
+		var receivedMap map[string]interface{}
+
+		response, err := Get(GetRequest{Path: url, User: UserRoles{Name: "admin", Roles: []string{"admin"}} })
+		if err != nil {
+			if !errorCase {
+				t.Errorf("Error %v received for Url: %s", err, url)
+			} else if expErr != nil {
+				checkErr(t, err, expErr[0])
+			}
+			return
+		}
+
+		err = json.Unmarshal([]byte(expectedRespJson), &expectedMap)
+		if err != nil {
+			t.Errorf("failed to unmarshal %v err: %v", expectedRespJson, err)
+		}
+
+		respJson := response.Payload
+		err = json.Unmarshal([]byte(respJson), &receivedMap)
+		if err != nil {
+			t.Errorf("failed to unmarshal %v err: %v", string(respJson), err)
+		}
+
+		if reflect.DeepEqual(receivedMap, expectedMap) != true {
+			t.Errorf("Response for Url: %s received is not expected:\n Received: %s\n Expected: %s", url, receivedMap, expectedMap)
+		}
+	}
+}
+
+func processGetRequestWithFile(url string, expectedJsonFile string, errorCase bool, expErr ...error) func(*testing.T) {
+	return func(t *testing.T) {
+		var expectedMap map[string]interface{}
+		var receivedMap map[string]interface{}
+
+		jsonStr, err := ioutil.ReadFile(expectedJsonFile)
+		if err != nil {
+			t.Errorf("read file %v err: %v", expectedJsonFile, err)
+		}
+		err = json.Unmarshal([]byte(jsonStr), &expectedMap)
+		if err != nil {
+			t.Errorf("failed to unmarshal %v err: %v", jsonStr, err)
+		}
+
+		response, err := Get(GetRequest{Path: url, User: UserRoles{Name: "admin", Roles: []string{"admin"}}})
+		if err != nil {
+			if !errorCase {
+				t.Errorf("Error %v received for Url: %s", err, url)
+			} else if expErr != nil {
+				checkErr(t, err, expErr[0])
+			}
+		}
+
+		respJson := response.Payload
+		err = json.Unmarshal([]byte(respJson), &receivedMap)
```

Comment:
unnecessary type cast `[]byte(respJson)` -- respJson is already `[]byte `

Reply:
Addressed as suggested

Reply:
Not addressed

Reply:
Done. Added the corrected file missing in previous commit.